### ISSUE TITLE
Cilium Gateway unexpectedly translates grpc-web requests to gRPC

### ISSUE THREAD

**Issue by nevivurn** [cilium/cilium#31933]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

1. Create a Gateway and attach HTTPRoutes.
2. Notice that Cilium has added the `envoy.filters.http.grpc_web` filter in the CiliumEnvoyConfig.

What happened:
- Cilium Gateway translates grpc-web requests to gRPC by default.
- If the backend service expects grpc-web requests (and intends to perform the translation itself), it fails due to malformed requests.
- This configuration is applied by default, and there seems to be no way to disable this short of patching the CiliumEnvoyConfig manually.

What I expected:
- Cilium gateway should pass through grpc-web request to the backend.
- This should not be the default configuration, or at least there should be a mechanism to disable this configuration.

### Cilium Version

`v1.15.4`

### Kernel Version

`Linux worker-3 6.1.78-talos #1 SMP Wed Feb 21 10:56:19 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`

### Kubernetes Version

`v1.29.2`

### Regression

I have not tested it, but I suspect services expecting grpc-web were working before 5541837f408c30fcf68e23db30caddea3c17adae, merged as part of #28654, first appearing in v1.15.0.

### Anything else?

As mentioned above, the relevant code was added in 5541837f408c30fcf68e23db30caddea3c17adae, and is still in the default branch as of writing: https://github.com/cilium/cilium/blob/076cc7e9dd401787aeb0c5b9048384708915bfc8/operator/pkg/model/translation/envoy_http_connection_manager.go#L34-L39

Reading the above code, I believe this also affects Ingresses, but I have not tested them.

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by chaunceyjiang** [comment]
Make sense

**Comment by chaunceyjiang** [comment]
@sayboras  Can we add a parameter to control whether to enable `grpc-web`?

**Comment by sayboras** [comment]
:wave: 

Sorry for late reply, I was thinking about this, and cross check with other Gateway API implementation (e.g. contour, envoy), it's not uncommon to have grpc-web filter by default.

**Comment by nico151999** [comment]
It might not be uncommon that other implementations also have the GRPC web filter enabled by default but I could not find anything about it in the Gateway API spec, so I guess it is implementation-specific by now. If one develops an application oneself it is usually not a big deal to directly accept GRPC requests in the backend instead of GRPC web but if one intends to use an existing application (e.g. Zitadel) that uses GRPC web in the backend this behaviour causes that the backend cannot handle the requests. It is debatable which behaviour should be the default (as long as it is not specified in the Gateway API spec) but I would suggest that some way of configuring this is a necessity. This has been suggested by a user in [another issue](https://github.com/cilium/cilium/issues/26941#issuecomment-1932528974) before. In certain use cases this simply keeps me from being able to use the Gateway API otherwise. I would be happy to read other opinions about this topic. Thanks!

**Comment by nevivurn** [comment]
Still relevant.

See also projectcontour/contour#4290. Contour, one of the examples with similar behavior mentioned above, also have an open issue with people presenting use-cases that are blocked by this default feature.

**Comment by nevivurn** [comment]
Still relevant for us

**Comment by nevivurn** [comment]
Still relevant. In case others run into the same issue, we have been working around this issue with a sidecar that repeatedly patches the ciliumenvoyconfig, like so: ([example](https://github.com/bacchus-snu/cd-manifests/blob/0babfe48f2542518c8045e84c3c2c1aad84c05ef/argocd/waiter/bacchus-vpn/statefulset.yaml#L45-L62))

```shell
                  kubectl patch \
                    -n cilium-system \
                    --type=json \
                    ciliumenvoyconfigs.cilium.io/cilium-gateway-web-gateway \
                    -p '[{"op":"remove","path":"/spec/resources/0/filterChains/0/filters/0/typedConfig/httpFilters/0"},{"op":"remove","path":"/spec/resources/0/filterChains/1/filters/0/typedConfig/httpFilters/0"}]'
```

The command needs to be adjusted to match the namespace/gateway name/number of listeners/etc.

**Comment by eduartua** [comment]
> It might not be uncommon that other implementations also have the GRPC web filter enabled by default but I could not find anything about it in the Gateway API spec, so I guess it is implementation-specific by now. If one develops an application oneself it is usually not a big deal to directly accept GRPC requests in the backend instead of GRPC web but if one intends to use an existing application (e.g. Zitadel) that uses GRPC web in the backend this behaviour causes that the backend cannot handle the requests. It is debatable which behaviour should be the default (as long as it is not specified in the Gateway API spec) but I would suggest that some way of configuring this is a necessity. This has been suggested by a user in [another issue](https://github.com/cilium/cilium/issues/26941#issuecomment-1932528974) before. In certain use cases this simply keeps me from being able to use the Gateway API otherwise. I would be happy to read other opinions about this topic. Thanks!

This issue is still relevant.
